ui/packages/atlasmap/src/UI/ConditionalExpressionInput.tsx

```diff
@@ -32,11 +33,12 @@ export const ConditionalExpressionInput: FunctionComponent<IConditionalExpressio
   mappingExpressionRemoveField,
   mappingExpression,
   onConditionalMappingExpressionEnabled,
+  onMappingHasSourceCollection,
   onToggleExpressionMode,
   trailerId,
 }) => {
   function onToggle() {
-    onToggleExpressionMode();
+    onToggleExpressionMode(onMappingHasSourceCollection());
```

Comment:
`onMappingHasSourceCollection` is an implementation details that would be better handled inside the `onToggleExpressionMode` function. Doing this would also remove a prop from this component.